File: MemeShareSwiftUI/MemeShareSwiftUI/model/KullaniciResponse.swift
//
//  KullaniciResponse.swift
//  MemeShareSwift
//
//  Created by Alperen Saraç on 31.08.2025.
//

import Foundation
struct KullaniciResponse: Decodable,Equatable {
    let success: Bool
    let message: String?     // ⬅️ optional
    let userId: Int?         // ⬅️ optional

    enum CodingKeys: String, CodingKey {
        case success
        case message
        case userId = "user_id"
    }
}

File: meme_share_flutter/lib/model/kullanici_response.dart
class KullaniciResponse {
  final bool success;
  final String message;
  final int userId;

  KullaniciResponse({
    required this.success,
    required this.message,
    required this.userId,
  });

  factory KullaniciResponse.fromJson(Map<String, dynamic> json) {
    return KullaniciResponse(
      success: json['success'],
      message: json['message'],
      userId: json['user_id'],
    );
  }
}
